processing/src/main/java/io/druid/query/DefaultQueryMetrics.java

```diff
@@ -66,15 +65,6 @@ public void queryType(QueryType query)
   }
 
   @Override
-  public void interval(QueryType query)
```

Comment:
What I'm concerned with is that data sources or intervals are not meaningful alone for join. A list of intervals without their data sources don't give any insight. Data sources without intervals might be more useful than intervals alone, but they are not enough to analyze query performance. 

However, I agree on intervals consume much space which is not necessary. As you said, duration will be more useful. So, what do you think about adding ```dataSourcesAndDurations()``` for join query? Also, I'm fine with keeping ```dataSources()```.